tests/suites/test_suite_pk.function

```diff
@@ -531,19 +531,25 @@ void pk_rsa_verify_ext_test_vec( data_t * message_str, int digest,
                                  digest, hash_result, hash_len,
                                  result_str->x, sig_len );
 
-    /* Mbed TLS distinguishes "invalid padding" from "valid padding but
-     * the rest of the signature is invalid". This has little use in
-     * practice and PSA doesn't report this distinction.
-     * In this case, PSA returns PSA_ERROR_INVALID_SIGNATURE translated
-     * to MBEDTLS_ERR_RSA_VERIFY_FAILED
-     */
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-    if( result == MBEDTLS_ERR_RSA_INVALID_PADDING &&
-        ret == MBEDTLS_ERR_RSA_VERIFY_FAILED )
-        TEST_EQUAL( ret, MBEDTLS_ERR_RSA_VERIFY_FAILED);
+    if( result == MBEDTLS_ERR_RSA_INVALID_PADDING )
+    {
+        /* mbedtls_pk_verify_ext() may return MBEDTLS_ERR_RSA_INVALID_PADDING
+         * error depending on which path was taken.
+         * If the PSA path is used, it won't because Mbed TLS
+         * distinguishes "invalid padding" from "valid padding but
+         * the rest of the signature is invalid". This has little use in
```

Comment:
Sorry for picking on grammar, but I don't think we should have a full stop here, the explanation introduced by the "because" is not complete yet - the reason is "and PSA doesn't report this distinction". While editing this comment, I think we might want to expand on why a non-PSA path might be taken while `USE_PSA` is active.

Here's a suggestion:
```
    /* Mbed TLS distinguishes "invalid padding" from "valid padding but
     * the rest of the signature is invalid". This has little use in
     * practice and PSA doesn't report this distinction.
     * In this case, PSA returns PSA_ERROR_INVALID_SIGNATURE translated
     * to MBEDTLS_ERR_RSA_VERIFY_FAILED.
     * However, currently `mbedtls_pk_verify_ext()` may use either the PSA or the
     * Mbed TLS API, depending on the PSS options used. So, it may return either
     * INVALID_PADDING or INVALID_SIGNATURE.
     */
```

Reply:
I'm ok with that, I'll update the comment